src/PageSections/_Private/Globals.hack

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAPIDoc\PageSections\_Private;
+
+/**
+ * @reviewer This is /intentionally/ extremely painfully ugly.
+ * Do not accept the PR like this.
+ * I am not happy with this approach.
+ *
+ * This filtering must take place in InterfaceSynopsis::getMarkdown().
+ * It reads out the scanned class directly.
+ * We should not remove the methods from the class, so getMarkdown() won't find them.
+ * That is an even worse solution.
+ *
+ * getMarkDown() cannot take arguments, because it is an <<__Override>>
+ * and the constructor cannot take any new arguments either, because it is a <<__ConsistentConstruct>>.
+ * We also can't add a method to the class `->setHidePrivateMethods()`, it's constructor is never called directly.
+ * It is called via the <<__ConsistentConstruct>> of PageSection.
+ *
+ * I would like to discuss a better way to hand this boolean to that method.
+ */
```

Comment:
I'm not familiar with all the details so this solution may not be the best (or may not work at all), but you could maybe split the `InterfaceSynopsis` class into 2 classes like `FullInterfaceSynopsis` and `NonPrivateInterfaceSynopsis` (with a common base class to cover the shared parts, or one could extend the other).

From what I remember, passing down the CLI flag to wherever we build the list of PageSections shouldn't be too hard, so you could conditionally include one or the other class in the list there.

Reply:
Or possibly you could create a shape/class with options (right now just the one option) and pass it down to all the PageSection classes (through that consistent constructor), even though the InterfaceSynopsis would be the only one using the options right now...

Reply:
I'd go with the second one. That seems to be cleaner right now.
If we have something in the future that is a drastic change from the default behavior, we might want to implement alternative classes.